packages/ZbayLite/src/renderer/containers/widgets/createUsernameModal/CreateUsername.tsx

```diff
@@ -33,17 +34,50 @@ const CreateUsernameModal = () => {
   const joinCommunityModal = useModal(ModalName.joinCommunityModal)
   const createCommunityModal = useModal(ModalName.createCommunityModal)
   const loadingCommunityModal = useModal(ModalName.loadingPanel)
+  const initializedCommunities = useSelector(identity.selectors.unregisteredCommunitiesWithoutUserIdentity)
+  const isInitializedCommunity = initializedCommunities.length
+
+  const [isCreateUserNameStarted, setisCreateUserNameStarted] = useState('undecided')
+  const [isRetryingRegistration, setIsRetryingRegistration] = useState(false)
+
+  const isConnected = useSelector(socketSelectors.isConnected)
+
+  const unregisteredCommunities = useSelector(identity.selectors.unregisteredCommunities)
+  const isUnregisteredCommunity = unregisteredCommunities.length
+  const isOwner = useSelector(communities.selectors.isOwner)
+
+  useEffect(() => {
+    if (isConnected && isInitializedCommunity && isCreateUserNameStarted === 'undecided') {
```

Comment:
I think it'd be better to use boolean or enum type here. 'undecided' is typo-fragile